lib/Controller/AdminController.php

```diff
@@ -543,6 +541,18 @@ public function link(string $emulated, string $circleId, string $singleId): Data
 	private function setLocalFederatedUser(string $emulated): void {
 		$user = $this->userSession->getUser();
 		$this->federatedUserService->setCurrentPatron($user->getUID());
+
+		if (strpos($emulated, '=') > 0) {
+			[$type, $id] = explode('=', $emulated, 2);
+			switch (strtolower($type)) {
+				case 'singleid':
```

Comment:
Maybe something to document somewhere? Not sure whether there's documentation where it fits - but at least I'd document the logic in the annotations of this function, or in the OCS controller.